request/request.go

```diff
@@ -0,0 +1,18 @@
+package request
+
+import "encoding/json"
+
+// Request represents the data of an incoming request.
+type Request struct {
+}
+
+// RequestFromJSON decodes a JSON payload into a Request.
+func RequestFromJSON(data []byte) (Request, error) {
```

Comment:
func name will be used as request.RequestFromJSON by other packages, and that stutters; consider calling this FromJSON